framework/src/functions/PiecewiseConstant.C

```diff
@@ -67,7 +67,13 @@ PiecewiseConstant::value(Real t, const Point & p)
   const unsigned len = functionSize();
   const Real toler = 1e-14;
 
-  // endpoint case
+  // endpoint cases
+  if ( (_direction == LEFT  && x < (1+toler) * domain(0)) ||
+       (_direction == RIGHT && x < (1-toler) * domain(0)) )
```

Comment:
If I understand correctly, this assigns `range(0)` to `x` values arbitrarily far to the "left" of `domain(0)` as long as `_direction==RIGHT`?  In other words, we want to interpolate values outside the domain to the left end-point value, not throw an error?